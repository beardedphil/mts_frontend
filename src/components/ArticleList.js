import React from 'react';
import Article from './Article';

const ArticleList = ({ articles }) => {
    return (
        <div className="article-list">
            {
                articles.length === 0 ? (
                    <p>No articles found.</p>
                ) : (
                    articles.map((article) => {
                        return (
                            <Article article={article} />
                        );
                    })
                )
            }
        </div>
    );
};

export default ArticleList;
