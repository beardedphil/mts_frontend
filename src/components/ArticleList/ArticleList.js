import React from 'react';
import Article from '../Article/Article';
import './ArticleList.scss';

const ArticleList = ({ articles = [] }) => {
    return (
        <div className="article-list" data-testid="article-list">
            {
                articles.length === 0 ? (
                    <p>No articles found.</p>
                ) : (
                    articles.map((article) => {
                        return (
                            <Article article={article} key={article.id}/>
                        );
                    })
                )
            }
        </div>
    );
};

export default ArticleList;
