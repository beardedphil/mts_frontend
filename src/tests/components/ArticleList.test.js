import React from 'react';
import { shallow } from 'enzyme';
import ArticleList from '../../components/ArticleList';
import articles from '../fixtures/articles';

test('should render ArticleList correctly with no articles', () => {
    const wrapper = shallow(<ArticleList articles={[]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ArticleList correctly with articles', () => {
    const wrapper = shallow(<ArticleList articles={ articles } />);
    expect(wrapper).toMatchSnapshot();
});
