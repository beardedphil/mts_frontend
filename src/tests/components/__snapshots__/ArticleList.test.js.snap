// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArticleList correctly with articles 1`] = `
<div
  className="article-list"
>
  <Article
    article={
      Object {
        "article_link": "https://slate.com/culture/2020/12/wonder-woman-1984-online-response-negative.html",
        "id": 421,
        "image_link": "https://compote.slate.com/images/9d1b35f7-1ecf-4315-850b-a3d654269bae.jpeg?width=780&height=520&rect=1560x1040&offset=0x0",
        "keywords": Array [
          "1984",
          "wonder",
          "comic",
          "response",
          "online",
          "ww84",
          "movies",
          "book",
          "woman",
          "future",
          "hbo",
          "blockbusters",
          "watch",
          "max",
          "movie",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "Wonder Woman 1984 online response: Is this the future of blockbusters?",
        "url": "http://localhost:8000/articles/421/",
      }
    }
    key="421"
  />
  <Article
    article={
      Object {
        "article_link": "https://slate.com/news-and-politics/2020/12/new-york-post-editorial-trump-concede.html",
        "id": 422,
        "image_link": "https://compote.slate.com/images/d0af0df8-d693-419d-a3ef-c89ed86aee54.jpeg?width=780&height=520&rect=615x410&offset=0x0",
        "keywords": Array [
          "outlet",
          "senate",
          "york",
          "editorial",
          "voter",
          "schemes",
          "fox",
          "conservative",
          "latest",
          "post",
          "trumps",
          "election",
          "board",
          "trump",
          "president",
          "reject",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "The New York Post Is the Latest Conservative Outlet to Reject Trump’s Election Schemes",
        "url": "http://localhost:8000/articles/422/",
      }
    }
    key="422"
  />
  <Article
    article={
      Object {
        "article_link": "https://slate.com/culture/2020/12/washington-football-team-bad.html",
        "id": 425,
        "image_link": "https://compote.slate.com/images/525420a4-92e3-49f9-9556-b4a02bb5a28f.jpeg?width=780&height=520&rect=2929x1953&offset=0x0",
        "keywords": Array [
          "theyre",
          "washington",
          "nfc",
          "football",
          "winning",
          "team",
          "teams",
          "season",
          "east",
          "2020",
          "good",
          "worst",
          "division",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "Washington Football Team, 2020 NFC East: The worst good season ever.",
        "url": "http://localhost:8000/articles/425/",
      }
    }
    key="425"
  />
  <Article
    article={
      Object {
        "article_link": "https://slate.com/culture/2020/12/wonder-woman-1984-online-response-negative.html",
        "id": 421,
        "image_link": "https://compote.slate.com/images/9d1b35f7-1ecf-4315-850b-a3d654269bae.jpeg?width=780&height=520&rect=1560x1040&offset=0x0",
        "keywords": Array [
          "1984",
          "wonder",
          "comic",
          "response",
          "online",
          "ww84",
          "movies",
          "book",
          "woman",
          "future",
          "hbo",
          "blockbusters",
          "watch",
          "max",
          "movie",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "Wonder Woman 1984 online response: Is this the future of blockbusters?",
        "url": "http://localhost:8000/articles/421/",
      }
    }
    key="421"
  />
  <Article
    article={
      Object {
        "article_link": "https://slate.com/news-and-politics/2020/12/new-york-post-editorial-trump-concede.html",
        "id": 422,
        "image_link": "https://compote.slate.com/images/d0af0df8-d693-419d-a3ef-c89ed86aee54.jpeg?width=780&height=520&rect=615x410&offset=0x0",
        "keywords": Array [
          "outlet",
          "senate",
          "york",
          "editorial",
          "voter",
          "schemes",
          "fox",
          "conservative",
          "latest",
          "post",
          "trumps",
          "election",
          "board",
          "trump",
          "president",
          "reject",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "The New York Post Is the Latest Conservative Outlet to Reject Trump’s Election Schemes",
        "url": "http://localhost:8000/articles/422/",
      }
    }
    key="422"
  />
  <Article
    article={
      Object {
        "article_link": "https://slate.com/culture/2020/12/washington-football-team-bad.html",
        "id": 425,
        "image_link": "https://compote.slate.com/images/525420a4-92e3-49f9-9556-b4a02bb5a28f.jpeg?width=780&height=520&rect=2929x1953&offset=0x0",
        "keywords": Array [
          "theyre",
          "washington",
          "nfc",
          "football",
          "winning",
          "team",
          "teams",
          "season",
          "east",
          "2020",
          "good",
          "worst",
          "division",
        ],
        "pub_date": "2020-12-29",
        "source": "slate",
        "title": "Washington Football Team, 2020 NFC East: The worst good season ever.",
        "url": "http://localhost:8000/articles/425/",
      }
    }
    key="425"
  />
</div>
`;

exports[`should render ArticleList correctly with no articles 1`] = `
<div
  className="article-list"
>
  <p>
    No articles found.
  </p>
</div>
`;
